### DISCUSSION TITLE
Explore/Loki: See logs for metrics queries

### DISCUSSION THREAD

**Discussion by ivanahuckova** [grafana/grafana#45587] [Observability]
**What would you like to be added**:
Based on discussion with @buro9 it would be a great idea to add option to loki's query editor to run simultaneously both - metrics and logs query. You would specify the metrics query and logs query would be created out of it. This way, you could visualise in graph results from metrics query and below you would have associated logs, which is very useful. 

Currently, you can use split views + sync times to do this, but this option/added feature would be much faster and more easier to use. 

@buro9 feel free to edit this description and add your thoughts. 🙂

**Comment by grafana-dee** [comment]
Example that came up in a usability test is that if you have a dashboard graph based on Loki that just shows the rate that the log is written to:
```rate({job="varlogs"}[1m])```

Then when you explore from the graph, Explore starts with `rate({job="varlogs"}[1m])` which means manually editing the Explore query every time to remove the rate function `{job="varlogs"}` so that within Explore the view will include the log panel below the graph - thus speeding up the process of exploring by providing more info.

**Comment by aocenas** [comment]
One idea that could solve is to have a sort of quick actions or hints supplied by the data source that can interact with explore. In this case Loki could provide `Add log query` button that would automatically add new query row with logs query based on the rate query. That would still better match current Explore functionality while adding new feature that plugins can use. Although it would also mean more work a bit of thought about the plugin API and what to allow plugins to do.

**Comment by wardbekker** [comment]
Would it make sense to make a combined metrics + logs view the default in Explore, and we just replace the logs histogram with a real metrics query so we don't get those partial histograms any more because of the 1000s limit?

**Comment by aocenas** [comment]
@wardbekker you should be able to do this, in the sense that you can write log and metrics query in explore and it will show both. Problem with the histogram inside the log panel is that it does some more things like it tries to set log level based on the message (parsing and checking if there is `error` for example) so not sure if you can do that with a metric query. Probably yes with the new regex parsing but it's not going to be something that people can write easily.

**Comment by ivanahuckova** [comment]
Implemented in https://github.com/grafana/grafana/pull/61864
